src/logging_manager/mod.rs

```diff
@@ -0,0 +1,340 @@
+        assert!(lines.next().unwrap().unwrap().ends_with("] ERROR test error"));
+        assert!(lines.next().unwrap().unwrap().ends_with("] INFO  test info"));
+        assert!(lines.next().unwrap().unwrap().ends_with("] TRACE test trace"));
+        assert!(lines.next().unwrap().unwrap().ends_with("] WARN  test warning"));
+        assert!(lines.next().is_none());
+
+        // clean up
+        remove_dir_all(logging_directory)?;
+        assert_eq!(false, logging_directory.exists());
+        Ok(())
+    }
+
+    #[test]
+    fn test_check_size() -> Result<(), std::io::Error> {
+
+        // random files and configs
+        let file1 = Path::new("2020_08_01.log");
+        let file2 = Path::new("2020_08_02.log");
+        let file3 = Path::new("2020_08_03.log");
+        let random_text = "Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.";
+        let config1 = LoggingConfig {
+            logging_directory: Path::new("./test_logs2/"),
+            max_dir_size: 0
+        };
+        let config2 = LoggingConfig {
+            logging_directory: Path::new("./test_logs3/"),
+            max_dir_size: 1000
+        };
+        let config3 = LoggingConfig {
+            logging_directory: Path::new("./test_logs4/"),
+            max_dir_size: 10000
+        };
+
+        // config 1: max dir size is 0, all files should be deleted
+        create_dir(&config1.logging_directory)?;
+        let mut file = File::create(&config1.logging_directory.join(&file1))?;
+        file.write_all(random_text.as_bytes())?;
+        file = File::create(&config1.logging_directory.join(&file2))?;
+        file.write_all(random_text.as_bytes())?;
+        file = File::create(&config1.logging_directory.join(&file3))?;
+        file.write_all(random_text.as_bytes())?;
+        file.sync_all()?;
+        check_size(&config1)?;
+        remove_dir(&config1.logging_directory)?;
+
+        // config 2: max dir size is 1000 bytes, first file should be deleted (second and third remains)
+        create_dir(&config2.logging_directory)?;
+        let mut file = File::create(&config2.logging_directory.join(&file1))?;
+        file.write_all(random_text.as_bytes())?;
+        file = File::create(&config2.logging_directory.join(&file2))?;
+        file.write_all(random_text.as_bytes())?;
+        file = File::create(&config2.logging_directory.join(&file3))?;
+        file.write_all(random_text.as_bytes())?;
+        file.sync_all()?;
+        check_size(&config2)?;
+        assert_eq!(false, config2.logging_directory.join(&file1).exists());
```

Comment:
Could you do assert!(!config2.logging_directory.join(&file1).exists())?

Reply:
Yeah it would evaluate !false = true

Reply:
Haha yeah I know. I was just wondering why you do that in some places and use assert! in other places.

Reply:
I find it personally more readable

Reply:
Makes sense